### ISSUE TITLE
Dataset proposal: officegraph

### ISSUE THREAD

**Issue by adamskas** [RiverBench/RiverBench#149]
### Description of the dataset

[OfficeGraph](https://github.com/RoderickvanderWeerdt/OfficeGraph/tree/main) is a real-world data of measurements from 444 IoT devices  taken over 11 months. The devices are made up of 17 different sensor models which measure different properties. Data was taken in Dutch 7-story office building and consists of about 90 Million RDF triplets. See also the [paper](https://2024.eswc-conferences.org/wp-content/uploads/2024/04/146640427.pdf).

### Value brought by the dataset

The dataset contains real, high volume data of IoT measurements expressed in SAREF ontology. SAREF ontology is not yet used by any benchmark at RiverBench and this dataset would bring. 

In this dataset the single element is a single measurement that captures one of a few properties (temperature, humidity, motion,  etc.) depending on the device that has taken that measurement. On the other hand in already existing benchmarks such as [assist-iot-weather](https://w3id.org/riverbench/datasets/assist-iot-weather/dev) all elements are structured the same way and elements does not differ in what type of information they capture.

### Proposed identifier

officegraph

### Link to the source of the dataset

https://github.com/RoderickvanderWeerdt/OfficeGraph

### License of the dataset

https://spdx.org/licenses/CC-BY-4.0.html

### Creator(s)

- [Roderick van der Weerdt](https://github.com/RoderickvanderWeerdt)
- Victor de Boer
- Ronald Siebes
- Ronnie Groenewold
- Frank van Harmelen

### Stream type (according to RDF-STaX)

RDF graph stream

### Stream element count

15M

### How were the stream elements split

An event will be a singular measurement taken by single device

Example of single element:
```
ic:measurement_R5_2__co2__10411 a saref:Measurement ;
    saref:hasTimestamp "2022-10-10T12:12:00"^^xsd:dateTime ;
    saref:hasValue "716"^^xsd:float ;
    saref:isMeasuredIn om:partsPerMillion ;
    saref:relatesToProperty ic:property_R5_2__co2_ ;
    ic:hasRoundedValue ic:rounded_value_716.0 ;
    ic:measuredAtTime ic:2022-10-10_120000 ;
    ic:next_node ic:measurement_R5_2__humidity__10411 ;
    ic:previous_node ic:measurement_R5_2__temp__10410 .

ic:rounded_value_716.0 ic:ofMeasurement ic:measurement_R5_2__co2__10411 .
ic:2022-10-10_120000 ic:measuredDuring ic:measurement_R5_2__co2__10411 .

ic:2022-10-10_120000 ic:atDate wd:Q69306580 .
```

Another example:
```
ic:measurement_SmartSense_Multi_Sensor_22__temp__1863 a saref:Measurement ;
    saref:hasTimestamp "2022-03-15T15:05:00"^^xsd:dateTime ;
    saref:hasValue "21.2"^^xsd:float ;
    saref:isMeasuredIn om:degreeCelsius ;
    saref:relatesToProperty ic:property_SmartSense_Multi_Sensor_22__temp_ ;
    ic:hasRoundedValue ic:rounded_value_21.2 ;
    ic:measuredAtTime ic:2022-03-15_150000 ;
    ic:next_node ic:measurement_SmartSense_Multi_Sensor_22__temp__1864 ;
    ic:previous_node ic:measurement_SmartSense_Multi_Sensor_22__temp__1862 .

ic:rounded_value_21.2 ic:ofMeasurement ic:measurement_SmartSense_Multi_Sensor_22__temp__1863 .
ic:2022-03-15_150000 ic:measuredDuring ic:measurement_SmartSense_Multi_Sensor_22__temp__1863 .

ic:2022-03-15_150000 ic:atDate wd:Q69306366 .
```

### Technical features

- [ ] Uses RDF-star
- [ ] Uses generalized triples
- [ ] Uses generalized datasets

### Additional technical notes (if any)

The dataset uses [SAREF](https://saref.etsi.org/) ontology which is a domain standard ontology specifically created to model measurements of different IoT devices. The creators also introduced six new subclasses of SAREF classes and two new RDF properties to enable a more detailed representation of the building data. 
Specific information about graph properties can be found in [original paper](https://2024.eswc-conferences.org/wp-content/uploads/2024/04/146640427.pdf).

### Metadata license

- [X] I agree to the terms

**Comment by Ostrzyciel** [comment]
Hi! Thank you for your submission.

Could you please clarify the following points?

- What will be the length of this stream? Please give at least a rough estimate.
- Could you give us an example of how a stream element would look? This will make it clearer how exactly you intend to split the dataset into elements.
- In the "Value brought by dataset" please be more specific and compare this dataset to what is already in RiverBench. What is different from [`assist-iot-weather`](https://w3id.org/riverbench/datasets/assist-iot-weather/dev)/[`assist-iot-weather-graphs`](https://w3id.org/riverbench/datasets/assist-iot-weather-graphs/dev) and [`citypulse-traffic`](https://w3id.org/riverbench/datasets/citypulse-traffic/dev)/[`citypulse-traffic-graphs`](https://w3id.org/riverbench/datasets/citypulse-traffic-graphs/dev)?

**Comment by Ostrzyciel** [comment]
Thank you for the updates!

**Comment by Ostrzyciel** [comment]
Thank you for the updates! I have only one last point: maybe we could name the dataset as "office-graph" to more accurately represent the original name?

@RiverBench/curators could you take a look and say if this is OK for you?

**Comment by Ostrzyciel** [comment]
Oh, and one more thing, the RDF stream type here is an RDF graph stream. Please correct that.

**Comment by anhlt18vn** [comment]
It is a nice source to add to the Riverbench

**Comment by jpcik** [comment]
Very relevant, and using a well known schema

**Comment by adamskas** [comment]
Hi @Ostrzyciel I have corrected the stream type and change the dataset name to "officegraph"

**Comment by Ostrzyciel** [comment]
> Hi @Ostrzyciel I have corrected the stream type and change the dataset name to "officegraph"

Perfect, thanks! Here is your repo: https://github.com/RiverBench/dataset-officegraph

You should now upload the dataset and fill out the metadata file using the instructions here: https://riverbench.github.io/v/dev/documentation/creating-new-dataset/#step-3-upload-the-dataset-sources

Let me know if you have any questions.

**Comment by adamskas** [comment]
Hi @Ostrzyciel I have completed the upload of dataset and filling of metadata

**Comment by Ostrzyciel** [comment]
@adamskas Thanks! It does work, and it shows up on the website now: https://riverbench.github.io/v/dev/datasets/officegraph/

But, there are a few issues:

- Under "creators" besides yourself you should also list (first and foremost) the creators of the original dataset. This is required by the CC BY 4.0 license.
- Please fill out the textual descriptions under `stax:hasStreamTypeUsage` and `rb:hasStreamElementSplit`. You already have the explanation above in the proposal template.
- Please include a short description of how the elements are ordered in the dataset description, in the metadata.
- Could you also add the `dcat:temporalResolution` property to the metadata? How often was the data sampled?
- I've also opened a PR with some more minor fixes – please review it and merge it if it's OK. https://github.com/RiverBench/dataset-officegraph/pull/1

**Comment by Ostrzyciel** [comment]
Oh, and please do remember to add a link to the device metadata dataset in the dataset description. :)

**Comment by adamskas** [comment]
Pushed new metadata file with fixes. 
The link to Zenodo metadata dataset will be added later but not later than end of the next week

**Comment by Ostrzyciel** [comment]
Yes, but the CI was not passing due to syntax errors in metada.ttl :) please always check if it does pass.

I fixed it for you: https://github.com/RiverBench/dataset-officegraph/commit/d451ff760bd1d8d1fd0b77ee5c7a079986b5fc66

Looks great, thank you for your submission! :) I will close the issue now. Please add the Zenodo link to the metadata when it becomes possible.
